### ISSUE TITLE
Introduce Liger Fused Cross Entropy Kernel to FOAK Plugin 

### ISSUE THREAD

**Issue by achew010** [foundation-model-stack/fms-acceleration#76]
## Description

Consider adding additional Fused CrossEntropyLoss Kernel to FOAK set of kernels

A comparison of the current [FOAK kernels](https://github.com/foundation-model-stack/fms-acceleration/tree/main/plugins/fused-ops-and-kernels) against the kernels from [Liger](https://github.com/linkedin/Liger-Kernel/tree/main) using Liger's [benchmark script](https://github.com/achew010/Liger-Kernel/tree/foak-benchmarks) with the following parameters;

- model: "meta-llama/Meta-Llama-3-8B"
- dataset: 10000 sample subset of "tatsu-lab/alpaca"
- max_seq_length: 512

![image](https://github.com/user-attachments/assets/c6145cd2-826e-4500-8f0c-37e1662ff8b5)

**Comment by fabianlim** [comment]
@achew010 currently we only install the non-chuncked CE loss from unsloth. It seems to be OK for llama3, but scaling up, we should consider handling chunking also

**Comment by anhuong** [comment]
Implemented (2) in this PR https://github.com/foundation-model-stack/fms-acceleration/pull/93 but as noted above it suffers from maintainability as
1. llama and mistral have very similar forward functions but mixtral has different forward function that also needs to be copied in and granite has a slight difference
2. hard to maintain between different transformers versions
3. was not able to show train_runtime improvement from other triton kernels but was able to show memory improvement (as noted in the PR)

Worked on implementing (3) via PR https://github.com/foundation-model-stack/fms-acceleration/compare/main...anhuong:fms-acceleration:fused-cross-entropyloss?expand=1 however the model patching rules aren't quite right here.

Thus it could be worth it to implement (2) via the PR for just transformers v4.44/4.43 and then implement the new solution with the model patching changes needed form #98.
